agents-common/src/main/java/org/apache/ranger/plugin/policyevaluator/RangerDefaultPolicyEvaluator.java

```diff
@@ -1512,4 +1504,27 @@ private List<RangerConditionEvaluator> createPolicyConditionEvaluators(RangerPol
 		return ret;
 	}
 
+	private static boolean isMatchForResourceMatchingScope(final RangerAccessRequest.ResourceMatchingScope scope, final RangerPolicyResourceMatcher.MatchType matchType, boolean isAnyMatch) {
+		boolean ret = false;
+		if (isAnyMatch){
+			ret = matchType !=  RangerPolicyResourceMatcher.MatchType.NONE;
+		}
+		else if (scope!=null) {
+				switch (scope) {
+						case SELF_OR_DESCENDANTS: {
+							ret = matchType != RangerPolicyResourceMatcher.MatchType.NONE;
```

Comment:
when scope is SELF_OR_DESCENDENTS, is it okay to return true when matchType is ANCESTOR?

Reply:
This would be out of scope for this PR.
SELF_OR_DESCENDANT ResourceMatchingScope is being used in many other places of the code. We would need more analysis and testing for making changes to this part of the code. This PR tries to keep current behavior as is.

Reply:
My point is - this PR introduces a regression in handing SELF_OR_DESCENDANT scope for ANCESTOR match. Please review and update.

Reply:
@fateh288  - my bad. The regression was for earlier comment. I agree that handling of ANCESTOR is outside the scope of this PR.